### ISSUE TITLE
tf.nn.softmax_cross_entropy_with_logits_v2() not warning incorrect input shapes when placeholders are provided

### ISSUE THREAD

**Issue by hakillha** [tensorflow/tensorflow#22270]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.10.0
- **Python version**: 3.5.2
- **GPU model and memory**: GTX1070

### Describe the problem
Following is a simple snippet to reproduce the problem. If I'm understanding this correctly, the shape of _labels_ is clearly wrong here but tensorflow doesn't warn you and when the same problem is present in my actual code, not even a runtime error will show up. Appreciate it if you guys can help me out on this especially if I mistook this. Otherwise this looks like a devastating issue.

### Source code / logs
a = tf.placeholder(tf.int32, (None))
b = tf.constant([[1.0,2,3],[4,5,6]])
c = tf.nn.softmax_cross_entropy_with_logits_v2(labels=a, logits=b)

**Comment by hakillha** [comment]
> The shape of `a` is unknown, see [tf.Dimension](https://www.tensorflow.org/api_docs/python/tf/Dimension). Could you clarify where is wrong?

If we are to provide placeholders for both the `label` and `logits` arguments, shouldn't they both have the shape (?, # of classes)? If for example in the above snippet `a` has only one dimension then there is no way this is an input with correct shape so shouldn't tensorflow warn about it?

**Comment by hakillha** [comment]
Okay in my actual code it was `sparse_softmax_cross_entropy_with_logits` that was run so I think there should be runtime error if the inputs for `softmax_cross_entropy_with_logits` are not compatible but no warning thing with the above snippet still persists which might be not that ideal a situation.
